crates/props-derive/src/lib.rs

```diff
@@ -0,0 +1,340 @@
+#![recursion_limit = "128"]
+extern crate proc_macro;
+extern crate quote;
+extern crate syn;
+
+use proc_macro::TokenStream;
+use proc_macro2::{Ident, Span};
+use quote::{quote, ToTokens};
+use std::error::Error;
+use std::iter;
+use syn::parse_macro_input;
+use syn::{
+    punctuated, DeriveInput, GenericParam, Generics, Meta, NestedMeta, Type, TypeParam, Visibility,
+    WhereClause,
+};
+
+struct PropField {
+    name: Ident,
+    ty: Type,
+    wrapped_name: Result<Ident, Box<dyn Error>>,
+}
+
+#[proc_macro_derive(Properties, attributes(props))]
+pub fn derive(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let props_name = input.ident;
+    let vis = input.vis;
+
+    let generics = input.generics;
+    let generic_params = &generics.params;
+    let generic_where = &generics.where_clause;
+    let generic_idents = {
+        let generic_idents = generics.params.iter().filter_map(|param| match param {
+            GenericParam::Type(TypeParam { ident, .. }) => Some(quote! { #ident }),
+            _ => unimplemented!(),
+        });
+
+        quote! {#(#generic_idents),*}
+    };
+
+    let named_fields = match input.data {
+        syn::Data::Struct(data) => match data.fields {
+            syn::Fields::Named(fields) => fields,
+            _ => unimplemented!(),
+        },
+        _ => unimplemented!(),
+    };
+
+    let mut prop_fields: Vec<PropField> = named_fields
+        .named
+        .into_iter()
+        .map(|field| {
+            let wrapped_name = check_required(&field);
+            PropField {
+                name: field.ident.unwrap(),
+                ty: field.ty,
+                wrapped_name,
+            }
+        })
+        .collect();
+
+    for pf in prop_fields.iter() {
+        match &pf.wrapped_name {
+            Err(err) => match err.downcast_ref::<syn::Error>() {
+                Some(err) => return TokenStream::from(err.to_compile_error()),
+                None => {}
```

Comment:
Is it okay to ignore errors other than syn::Error? It would be good to have a comment pointing out that we ignore all other errors and why.